build_bins.sh

```diff
@@ -94,7 +96,7 @@ EOF
 # defaults.
 #
 pushd $DIR/vendor > /dev/null
-/opt/chefdk/bin/chef-client -z -r 'recipe[bach_repository]' -c $TMPFILE
+/opt/chefdk/bin/chef-client -z -r 'recipe[bcpc::bach_repository_wrapper],recipe[bach_repository]' -c $TMPFILE
```

Comment:
i think we should do `include_recipe 'bach_repository'` inside `bcpc::bach_repository_wrapper` for reasons i posted in other comments.

Reply:
So we use the wrapper recipe as a wrapper recipe for more than [`build_bins`](https://github.com/bloomberg/chef-bach/pull/1206/files#diff-241f80a9f414eb8c74ac43a71e8570e7R99). We also use `bach_repository_wrapper` in [`recipe[bcpc::chef_vault_download]`](https://github.com/bloomberg/chef-bach/pull/1206/files#diff-12a3fefb1f76b03d12b9f4e3d4636874) which does not then call `recipe[bach_repository::default]` but instead calls `recipe[bach_repository::gems]`. This would require us calling all of bach_repositroy every Chef run we download Vault. Also we use `recipe[bach_respository_wrapper]` when we then call [`recipe[bach_repository::apt]`](https://github.com/bloomberg/chef-bach/pull/1206/files#diff-89d187ed8768ffde26c53a9c5a96e17eR62) in the Chef-Server.

This is a wrapper before we call into any point in bach_respository -- it is not an entrypoint to `recipe[bach_respository::default]` as such it would be in appropriate to force a specific bach_repository entry point at this time.

Reply:
I guess this is another set of implicit dependencies that this PR introduces.  After this PR, most of us will forget that this relationship exists and will take a lot of time for us to debug what is going on.